core/Updates/3.0.0-b2.php

```diff
@@ -21,4 +23,84 @@ public static function isMajorUpdate()
     {
         return true;
     }
+
+    /**
+     * @var MigrationFactory
+     */
+    private $migration;
+
+    /**
+     * @var string
+     */
+    private $userTable = 'user';
+
+    public function __construct(MigrationFactory $factory)
+    {
+        $this->migration = $factory;
+    }
+
+    /**
+     * Here you can define one or multiple SQL statements that should be executed during the update.
+     * @param Updater $updater
+     * @return Migration[]
+     */
+    public function getMigrations(Updater $updater)
+    {
+        $migrations = [];
+        $migrations = $this->getUserDatabaseMigrations($migrations);
+
+        return $migrations;
+    }
+
+
+    public function doUpdate(Updater $updater)
+    {
+        $updater->executeMigrations(__FILE__, $this->getMigrations($updater));
+
+        $this->hashUserPasswords();
+    }
+
+      /**
+     * @param Migration[] $queries
+     * @return Migration[]
+     */
+    private function getUserDatabaseMigrations($queries)
+    {
+        $queries[] = $this->migration->db->changeColumn($this->userTable, 'password', 'password', 'VARCHAR(255) NOT NULL');
+        $queries[] = $this->migration->db->addColumn($this->userTable, 'password_hashed', 'VARCHAR(255) NOT NULL', 'password');
```

Comment:
BTW: We would then need a new method `generateTokenAuth` but this would be likely not needed in the API :) `getTokenAuth()` would then do what it actually says meaning returning a token, not generating it :)